### ISSUE TITLE
Adding the support for OneDrive WebDav (personal and family plan without sharepoint)

### ISSUE THREAD

**Issue by kurokawaikki** [alist-org/alist#5239]
### Please make sure of the following things

- [X] I have read the [documentation](https://alist.nn.ci).
- [X] I'm sure there are no duplicate issues or discussions.
- [X] I'm sure this feature is not implemented.
- [X] I'm sure it's a reasonable and popular requirement.

### Description of the feature / 需求描述

Currently, the documentation of onedrive only supports the E3 E5 accounts with share points or azure.
However, if we could also link the onedrive of personal or family plan will be great.
Do you have any clue to do it?
Thank you very much.

### Suggested solution / 实现思路

I found the webdav method stated here: https://winscp.net/eng/docs/guide_microsoft_onedrive

I tried this method with other webdav applications and it worked.
However, when I add the info to Alist webdav, it will not work...
When I access the folder, it will show: failed get objs: failed to list objs: PROPFIND /: 200
Thank you very much.

**Comment by itsHenry35** [comment]
can't you just use [this way](https://alist.nn.ci/guide/drivers/onedrive.html) to add OneDrive driver?

**Comment by kurokawaikki** [comment]
I tried that but it needs azure from organization.
Therefore, I would like to try the webdav method.
Are you possible to link onedrive?

**Comment by xhofe** [comment]
Which step was unsuccessful on using https://alist.nn.ci/guide/drivers/onedrive.html?
I use my personal onedrive account with it and it works fine.

![image](https://github.com/alist-org/alist/assets/36558727/17b45397-8d11-4b0e-b33d-52552f244557)

**Comment by kurokawaikki** [comment]
I try to register the Free Azure trial with my personal account.
And I followed the steps in the documentation.
Finally, I found the error when I tried to get the refresh token.
Here is the error message:
`unauthorized_client: The client does not exist or is not enabled for consumers. If you are the application developer, configure a new application through the App Registrations in the Azure Portal at https://go.microsoft.com/fwlink/?linkid=2083908.`
Thank you very much.
